src/forgivingJoin.js

```diff
@@ -0,0 +1,192 @@
+        }
+    });
+    return {
+        count: count,
+        keys: keys
+    };
+};
+
+/** Indexes CSV data by given column name
+ * @param {Object[]} data - The data to be indexed
+ * @param {String} key - The column name to index by
+ * @return {Object<String, Object>} The rows in `data` indexed by the values in column `key`
+ */
+fluid.data.indexByColumn = function (data, key) {
+    var togo = {};
+    data.forEach(function (row) {
+        togo[row[key]] = row;
+    });
+    return togo;
+};
+
+/** Copies data from a source object into a target object applying a period-separated prefix to each member name
+ * @param {Object} target - The object to receive the new properties. *This will be modified by the call*
+ * @param {Object} source - The source object from which all properties will be copied
+ * @param {String} prefix - The prefix which will be applied to each property name, separated from it by a period.
+ */
+fluid.data.copyWithPrefix = function (target, source, prefix) {
+    fluid.each(source, function (value, key) {
+        target[prefix + "." + key] = source[key];
+    });
+};
+
+/** Undoes the prefixing effect of `fluid.data.copyWithPrefix' and extracts the period-separated prefix from a property name
+ * @param {String} prefixedKey - The property name for which the prefix and key is to be extracted
+ * @return {Object} A structure containing members:
+ *     {String} dataset The prefix in the supplied key
+ *     {String} key The suffix in the supplied key
+ */
+fluid.data.parsePrefixedKey = function (prefixedKey) {
+    var dotpos = prefixedKey.indexOf(".");
+    return {
+        dataset: prefixedKey.substring(0, dotpos),
+        key: prefixedKey.substring(dotpos + 1)
+    };
+};
+
+/** The operation of `fluid.getMembers` but for a structure with two levels of nesting
+ * @param {Array|Object} holder - The doubly nested container to be filtered
+ * @param {String|String[]} path - An EL path to be fetched from each nested-level member
+ * @return {Array|Object} - The desired structure of fetched members
+ */
+fluid.getMembersDeep = function (holder, path) {
+    return fluid.transform(holder, function (oneHolder) {
+        return fluid.getMembers(oneHolder, path);
+    });
+};
+
+fluid.forgivingJoin = function (record, datasets) {
+    var left = datasets[record.left].value;
+    var right = datasets[record.right].value;
+
+    var leftKeys = fluid.data.dataToKeys(left);
+    var rightKeys = fluid.data.dataToKeys(right);
+
+    var intersects = [];
+    fluid.each(leftKeys, function (leftHash, leftKey) {
+        fluid.each(rightKeys, function (rightHash, rightKey) {
+            var record = fluid.data.intersect(leftHash, rightHash);
+            record.leftKey = leftKey;
+            record.rightKey = rightKey;
+            intersects.push(record);
+        });
+    });
+    intersects.sort(function (inta, intb) {
+        return intb.count - inta.count;
+    });
+    var best = intersects[0];
```

Comment:
Using the best matched column assumes only one single column serves as the primary key to join 2 datasets. It doesn't cover the use case of the composite primary key where more than one columns are used to form the primary key to identify matched records in 2 datasets.

Reply:
Thanks for the suggestion - I've updated the README to explain the algorithm used and written up the improvement as https://issues.fluidproject.org/browse/DATA-1